lib/THC/THCTensorIndex.cu

```diff
@@ -362,48 +372,90 @@ void THCudaTensor_indexSelect(THCState *state, THCudaTensor *res, THCudaTensor *
   THArgCheck(srcDims > 0, 2, "Source tensor is empty");
 
   THLongStorage *newSize = THCudaTensor_newSizeOf(state, src);
-  THLongStorage_set(newSize, dim, nIndex);
-  THCudaTensor_resize(state, res, newSize, NULL);
+  THLongStorage_set(newSize, dim, numIndices);
+  THCudaTensor_resize(state, dst, newSize, NULL);
   THLongStorage_free(newSize);
 
-  if (THCudaTensor_isContiguous(state, src) &&
-      THCudaTensor_isContiguous(state, res) &&
-      THCudaTensor_isContiguous(state, indices) &&
-      dim == 0)
-  {
-    long stride = THCudaTensor_stride(state, src, 0);
+  int indContig = THCudaTensor_isContiguous(state, indices);
 
-    int blockX = std::min(THCCeilDiv(nIndex, 4L), 65535L);
-    int blockY = std::min(THCCeilDiv(stride, 128L), 65535L);
+  // The `src` is partitioned into two parts:
+  // -the number of slices over which we are indexing, which is the
+  // total size of the tensor ignoring dimension `dim`;
+  // -the number of indices we are indexing over each of those slices,
+  // which is the total size of the tensor `indices`
+  long totalSize = THCudaTensor_nElement(state, dst);
+  long numSlices = totalSize / numIndices;
```

Comment:
Would `sliceSize` be a better name? When I see `numSlices`, I feel it should be the same as `numIndices`.